### DISCUSSION TITLE
Spoiler plugin: How to format shortcode data as markdown

### DISCUSSION THREAD

**Discussion by vapaamies** [getnikola/nikola#3793] [Q&A]
Hi,

I need spoiler plugin for Nikola, like some forums have. It's the easiest to add it via simple template-based shortcode, but... how to compile data between `{{% spoiler}}` and `{{ /spoiler }}` using markdown compiler? I use markdown for my site. Now it outputs spoiler body as is.

Or, it is easier to add two shortcodes – for opening and closing spoiler tag? Will it work?

And, what about nested spoilers?

**Comment by Kwpolska** [comment]
Nesting shortcodes is very likely impossible. Compiling Markdown might be possible with a Python-based plugin that calls a Markdown compiler.

**Reply by vapaamies** [reply]
> Compiling Markdown might be possible with a Python-based plugin that calls a Markdown compiler.

Yes, cap! I need a code sample. I've already made some custom improvements in my fork of Nikola (visible in the profile), but can't spend much time to dive into Nikola internals myself, so need a tip.

> I doubt the current Nikola implementation can deal with nesting.

Okay, let's forget about nesting. Just a compiler call.

**Reply by Kwpolska** [reply]
You can get the `markdown` plugin object and call `compile_string` - see the `rst2html` command for an example: https://github.com/getnikola/nikola/blob/master/nikola/plugins/command/rst2html/__init__.py#L47-L54

**Comment by ralsina** [comment]
Actually I wrote a C library that parses shortcodes and supports nested
ones. Would need to write a cffi to use it from python and lots of work to
make it usable :-(

https://github.com/ralsina/shortcodes

**Reply by Kwpolska** [reply]
It is certainly possible in the technical sense, but I doubt the current Nikola implementation can deal with nesting.
